simpleclient_common/src/test/java/io/prometheus/client/exporter/common/ExemplarTest.java

```diff
@@ -0,0 +1,402 @@
+        .name("no_labels")
+        .help("help")
+        .quantile(0.5, 0.01)
+        .register(registry);
+    for (int i=1; i<=11; i++) { // median is 5
+      noLabelsDefaultExemplar.observe(i);
+    }
+    // Summaries don't have Exemplars according to the OpenMetrics spec.
+    assertNewFormat("no_labels{quantile=\"0.5\"} 5.0\n");
+    assertNewFormat("no_labels_count 11.0\n");
+    assertNewFormat("no_labels_sum 66.0\n");
+
+    assertOldFormat("no_labels{quantile=\"0.5\",} 5.0\n");
+    assertOldFormat("no_labels_count 11.0\n");
+    assertOldFormat("no_labels_sum 66.0\n");
+  }
+
+  @Test
+  public void testSummaryLabels() throws IOException {
+    Summary labelsNoExemplar = Summary.build()
+        .name("labels")
+        .help("help")
+        .labelNames("label")
+        .quantile(0.5, 0.01)
+        .register(registry);
+    for (int i=1; i<=11; i++) { // median is 5
+      labelsNoExemplar.labels("test").observe(i);
+    }
+    // Summaries don't have Exemplars according to the OpenMetrics spec.
+    assertNewFormat("labels{label=\"test\",quantile=\"0.5\"} 5.0\n");
+    assertNewFormat("labels_count{label=\"test\"} 11.0\n");
+    assertNewFormat("labels_sum{label=\"test\"} 66.0\n");
+
+    assertOldFormat("labels{label=\"test\",quantile=\"0.5\",} 5.0\n");
+    assertOldFormat("labels_count{label=\"test\",} 11.0\n");
+    assertOldFormat("labels_sum{label=\"test\",} 66.0\n");
+  }
+
+  private static class TestExemplarSampler implements CounterExemplarSampler, HistogramExemplarSampler {
+
+    private final String traceId;
+    private final String spanId;
+    private final long timestamp;
+
+    private TestExemplarSampler(String traceId, String spanId, long timestamp) {
+      this.traceId = traceId;
+      this.spanId = spanId;
+      this.timestamp = timestamp;
+    }
+
+    @Override
+    public Exemplar sample(double increment, Exemplar previous) {
+      return new Exemplar(increment, timestamp, TRACE_ID, traceId, SPAN_ID, spanId);
+    }
+
+    @Override
+    public Exemplar sample(double value, double bucketFrom, double bucketTo, Exemplar previous) {
+      return new Exemplar(value, timestamp, TRACE_ID, traceId, SPAN_ID, spanId);
+    }
+  }
+
+  private void assertOldFormat(String line) throws IOException {
+    StringWriter writer = new StringWriter();
+    TextFormat.write004(writer, registry.metricFamilySamples());
+    String metrics = writer.toString();
+    Assert.assertTrue("Line not found. Expected metric:\n" + line + "Actual metrics:\n" + metrics,
+        metrics.contains("\n" + line));
+  }
+
+  private void assertNewFormat(String line) throws IOException {
+    StringWriter writer = new StringWriter();
+    TextFormat.writeOpenMetrics100(writer, registry.metricFamilySamples());
+    String metrics = writer.toString();
+    Assert.assertTrue("Line not found. Expected metric:\n" + line + "Actual metrics:\n" + metrics,
+        metrics.contains("\n" + line));
+  }
+
+  private String timestampString() {
```

Comment:
Why are you creating a copy of this function in the tests?

Reply:
Removed the copy and used the version from `TextFormat`.